#! /usr/bin/env python
# Forearm camera configuration

PACKAGE='otto_sensors'
import roslib; roslib.load_manifest(PACKAGE)

from driver_base.msg import SensorLevels
from dynamic_reconfigure.parameter_generator import * 

gen = ParameterGenerator()
#       Name                    Type      Reconfiguration level             Description                                                  Default    Min   Max
gen.add("angle_min",		double_t,    SensorLevels.RECONFIGURE_STOP,    "The minimum angle to check in laser frame, offset from from the beginning of the scan.",	                  0.0,         0.0,   6.28318531)
gen.add("angle_width",        	double_t,    SensorLevels.RECONFIGURE_STOP,    "The width of laserscan beams to check in radians.", 	                  0.0,         0.0,   6.28318531)
gen.add("thresh_dist",        	double_t,    SensorLevels.RECONFIGURE_STOP,    "The distance to set points under this level to max_dist + 1.", 	  0.0,         0.0,   6.0)
gen.add("reflect_angles",	bool_t,      SensorLevels.RECONFIGURE_STOP,    "Whether or not to reflect min_angle and max_angle to also check the opposite side of the scan.", False)                                     

exit(gen.generate(PACKAGE, "laserscan_threshold", "ScanThresh"))